storage/storage_model.go

```diff
@@ -0,0 +1,105 @@
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/logkeeper/model"
+	"github.com/pkg/errors"
+)
+
+const metadataFilename = "metadata.json"
+
+// LogChunkInfo describes a chunk of log lines stored in pail-backed offline
+// storage.
+type LogChunkInfo struct {
+	BuildID  string
+	TestID   string
+	NumLines int
+	Start    time.Time
+	End      time.Time
+}
+
+func (info *LogChunkInfo) key() string {
+	var prefix string
+	if info.TestID != "" {
+		prefix = testPrefix(info.BuildID, info.TestID)
+	} else {
+		prefix = buildPrefix(info.BuildID)
+	}
+	return fmt.Sprintf("%s%d_%d_%d", prefix, info.Start.UnixNano(), info.End.UnixNano(), info.NumLines)
+}
+
+func (info *LogChunkInfo) fromKey(path string) error {
```

Comment:
Should this be a static function called logChunkFromKey instead? Or is populating an existing object a better pattern?

Reply:
Ah this C style initialization avoids some weirdness with returning pointers, doesn't it?

Reply:
It's a pattern we use a bunch. I don't know that there's an advantage besides keeping the package namespace clear.
I'm fine either way.